pkg/normalize/validate.go

```diff
@@ -553,6 +555,35 @@ func checkDuplicates(records []*models.RecordConfig) (errs []error) {
 	return errs
 }
 
+func uniq(s []uint32) []uint32 {
+	seen := make(map[uint32]struct{})
+	var result []uint32
+
+	for _, k := range s {
+		if _, ok := seen[k]; !ok {
+			seen[k] = struct{}{}
+			result = append(result, k)
+		}
+	}
+	return result
+}
+
+func checkLabelHasMultipleTTLs(records []*models.RecordConfig) (errs []error) {
+	m := make(map[string][]uint32)
+	for _, r := range records {
+		label := fmt.Sprintf("%s %s", r.GetLabelFQDN(), r.Type)
+
+		// if we have more records for a given label, append their TTLs here, but deduplicate slice before returning it
+		m[label] = uniq(append(m[label], r.TTL))
+
+		// if after the uniq() pass we still have more than one ttl, it means we have multiple TTLs for that label
+		if len(m[label]) > 1 {
+			errs = append(errs, Warning{fmt.Errorf("multiple TTLs detected for: %s. This should be avoided.", label)})
+		}
+	}
```

Comment:
I noticed that if there are many records each with unique TTLs, the warning message appears n-1 times.  This might be by design.  It might be better to have one warning per label+rtype.

To do that, I suggest 2 loops.  The first one does builds up the lists of TTLs per label+rtype; the second calls uniq() on the list and generates the warning if len>1.

Here's my test:

```
$ git diff
...
+  //
+    A("testttl", IP_MAIN, TTL(400)),
+    A("testttl", IP_MAIN, TTL(401)),
+    A("testttl", IP_MAIN, TTL(402)),
+    A("testttl", IP_MAIN, TTL(403)),
+  //
...

$ dnscontrol preview --domains whatexit.org
2022/04/29 15:41:56 printIR.go:88: 3 Validation errors:
2022/04/29 15:41:56 printIR.go:91: WARNING: multiple TTLs detected for: testttl.whatexit.org A. This should be avoided.
2022/04/29 15:41:56 printIR.go:91: WARNING: multiple TTLs detected for: testttl.whatexit.org A. This should be avoided.
2022/04/29 15:41:56 printIR.go:91: WARNING: multiple TTLs detected for: testttl.whatexit.org A. This should be avoided.
```

Reply:
good one, I'll change it!